### PR TITLE
Refs #32800 -- Improved CsrfViewMiddleware test coverage.

### PR DESCRIPTION
**Pull request by cjerdonek**
https://code.djangoproject.com/ticket/32800

This adds and improves the `CsrfViewMiddleware` tests, prior to fixing the ticket.

### CODE REVIEW

**Comment by felixxm** [comment]
@cjerdonek Thanks :+1: Can you resolve conflicts? (we merged #14518)

**Comment by cjerdonek** [comment]
Thanks, @felixxm. Conflicts resolved.

File: tests/csrf_tests/tests.py [hunk 1]
```diff
@@ -17,6 +18,53 @@
...
+class CsrfFunctionTests(SimpleTestCase):
+
+    def test_unmask_cipher_token(self):
+        cases = [
+            (TEST_SECRET, MASKED_TEST_SECRET1),
+            (TEST_SECRET, MASKED_TEST_SECRET2),
+            (32 * 'a', 'vFioG3XOLyGyGsPRFyB9iYUs341ufzIEvFioG3XOLyGyGsPRFyB9iYUs341ufzIE'),
+            (32 * 'a', 64 * 'a'),
+            (32 * 'a', 64 * 'b'),
+            (32 * 'b', 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb'),
```
Comment (apollo13) [inline tests/csrf_tests/tests.py:37]: I wonder if this and below wouldn't be more readable as `32 * 'a' + 32 * 'b'` but no strong feelings

**Review by apollo13 — APPROVED** [review]
LGTM

File: tests/csrf_tests/tests.py (hunk 1 above)
Comment (cjerdonek) [reply tests/csrf_tests/tests.py:37]: Thanks, @apollo13. That would be okay with me.

**Comment by felixxm** [comment]
@cjerdonek Thanks :+1: I reorganized commits a bit.
@apollo13 Thanks for the review :+1:

File: tests/csrf_tests/tests.py (hunk 1 above)
Comment (felixxm) [reply tests/csrf_tests/tests.py]: Changed.

**Pull request merged**